### PR TITLE
feat(icons): gallery for SVG icons

### PR DESCRIPTION
**Pull request by theJohnnyMe**
**Describe pull-request**  
Adding new tab called "Icons" for display of SVGs

**Solving issue**  
Fixes: AB#1549

**How to test**  
1. Run Aplifly link from down below
2. Open Foundation -> Icons -> Tab Icon
3. Check downloading single icon (it should be svg file)
4. Check downloading multiple icons (it should be zip file)
5. Check downloading all icons (it should be zip file)
6. Check search
7. Check downloading from modal (open single icon)
8. Check text on-page for typos/wrong grammar etc.

### CODE REVIEW

**Comment by theJohnnyMe** [comment]
@gaonita - Please check the code when you can
@alexlioliopoulos and @areblk - Please check options under "How to test" and whatever else you find. Ping me or change directly yourself (if possible) in CMS - It won't reflect automatically. but I can update PR with your changes.

**Comment by areblk** [comment]
Used preview link in Edge.

# Input on /foundations/icons
## (Enhancements)
1. Icon grid sizes present from largest to smallest in text and table, but image present smallest to largest. It should be presented in the same order.

2. What is 'established industry standards'? Does everyone in Scania know what this is, or is it a more general reference? I think this could be clarified.

3. What does 'Four keyshapes deriving from the Scania logo structure can be used as guides to create a consistent UI Icon Kit.' mean? Spelling error, four supposed to be for?

# /foundations/icons/icons

4. Search field doesn't show results on searches with blank space in them, like 'arrow down' or 'arrow '

6. Selecting and downloading only 1 icon still saves this to my computer as a .zip

7. Icons in downloaded .zip are empty and show error when I open file: 
_"This page contains the following errors:
error on line 1 at column 1: Document is empty
Below is a rendering of the page up to the first error."_

## Enhancements

7. At this page, we spell 'color' the american way, but on other pages we use the Brittish English spelling 'colour'. I think we should be consistent and stick to one type of English in the docs and in code.

8. I'd like to have an inline code example here. Should icons always be in a <span> element? How to size them? Will this be explained in a later code-tab?

9. Individual icons are very large in the gallery. How about scaling them down to 75 % or 50 % so that its easier for users to browse through. 

10. Trim input in 'search for icon' since it gives an incorrect result on 'arrow '

11. When using the search bar to narrow down icons, clicking 'Select all' selects all 97 icons even though it was probably not the users intent to do so. When a search result is found, perhaps let 'Select all' only select icons that are shown, or give the user the option to 'Select search results' or something.

12. Icons would preferably sort from A-Z, so that 'Calendar' show before 'Calendar Inactive'? This would place the more general icon before its variants

13. Modal for downloading icons always show vertical scroll bar even when icons are fewer than 4 (max height). 

uneccessary scrollbar

14. When selected icons are downloaded to my computer, modal should change to reflect this to the user. Right now, the modal looks the same before and after downloading is completed which may be confusing. 

15. Modal should show sum of file size for icons selected for download.

16. Next to 'Select all'-option, there could be a 'Unselect all'-option.

17. When 'Select all'- is active, the text should change to 'Unselect all' because that is what happens when I click again.

18. Whens icons are selected, the 'Download selected'-button has -sm class. In the modal, the button class is default. Both are used in narrow spaces so both buttons should use the same class, in my opinion.

19. Modal could have a 'Cancel'-button that will close the modal.

20. Modal should always have the same height, independently on how many icons are to be downloaded.

## Questions

21. When no icons are in modal for downloading, the button is disabled. But when on hover, button changes to blue. Still not possible to interact with it, but is this the intention - that color still changes on hover also for disabled button? Compare to how Bootstrap disabled buttons doesn't change color on hover, https://getbootstrap.com/docs/4.0/components/buttons/#disabled-state

**Review by alexlioliopoulos — APPROVED** [review]

**Review by areblk — APPROVED** [review]
OK to merge for me, but here are some feedback.

Used preview both in Edge and Firefox.

- 4, 6, 9, 17 fixed
- 5, 16 obsolete
- Other topics will be included in Azure

New:
23. Can we have that nice "copy" icon to the right in the url field? Compare to code snippets in code tabs.
24. To adhere to our own Do's and Don'ts on button texts, 'Zip listed & Download' can shorten to just 'Download'. That download will be in a .zip file is described in the footer text.
25. 'Download icon as SVG' can shorten to 'Download' and file format presented outside of button (if it's important)

26. Weird responsive behaviour on medium size viewports of <div> for search field and checkbox 'Select all' 
@ >= 672 px wide - checkbox is indented 
@ >= 800 px wide - large space between search field and checkbox
@ >= 992 px wide - checkbox label divides on two rows and the whole content area is narrowed down
@ >= 1055 px, side menu appears and rest of layout looks like for >991 px
@ >= 1184 px wide - checkbox div is lower than search field div
@ >= 1312 pix wide, anchor menu appears
@ >= 1376 px wide, huge gap between search field and checkbox

27. Horizontal line should always be as long as below columns

**Comment by theJohnnyMe** [comment]
@areblk  - Thanks for the check and new findings. I think I have managed to improve the grid and search+checkbox next to it. Regarding Code snippet copy - there is separate component for it which will need some adjustment to work in modal so lets leave it for future improvement - but I like idea, we should do it in next iteration ;)

**Comment by areblk** [comment]
> @areblk - Thanks for the check and new findings. I think I have managed to improve the grid and search+checkbox next to it. Regarding Code snippet copy - there is separate component for it which will need some adjustment to work in modal so lets leave it for future improvement - but I like idea, we should do it in next iteration ;)

"Select all" is still on two rows between 1184 and 1375 px, but all else looks good. OK with me!

File: src/app/components/new-icon-list/new-icon-list.component.scss [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+      right: -10px;
+    }
+  }
+
+  &:hover {
+    cursor: pointer;
+  }
+
+  sdds-icon {
+    //FIXME: Add correct rem
```
Comment (gaonita) [inline src/app/components/new-icon-list/new-icon-list.component.scss:39]: what does this mean? correct rem for the font size?

**Review by gaonita — APPROVED** [review]
Looks good to me!

File: src/app/components/new-icon-list/new-icon-list.component.scss (hunk 1 above)
Comment (theJohnnyMe) [reply src/app/components/new-icon-list/new-icon-list.component.scss:39]: Yes, I think it is inherited from the old pictograms file so the comment came from there.

**Pull request merged**